client_end_user/shared/src/commonMain/kotlin/presentation/profile/ProfileScreenModel.kt

```diff
@@ -0,0 +1,55 @@
+package presentation.profile
+
+import cafe.adriel.voyager.core.model.coroutineScope
+import domain.entity.UserDetails
+import domain.usecase.IManageUserUseCase
+import kotlinx.coroutines.CoroutineScope
+import presentation.base.BaseScreenModel
+import presentation.base.ErrorState
+import presentation.main.MainScreenUiEffect
+
+class ProfileScreenModel(
+  private val  manageUser: IManageUserUseCase
+) :BaseScreenModel<ProfileUIState, ProfileUIEffect >(ProfileUIState()) ,ProfileInteractionListener{
+
+    override val viewModelScope: CoroutineScope = coroutineScope
+
+    init {
+        getUserProfile()
+    }
+
+    private fun getUserProfile () {
+        tryToExecute(
+            { manageUser.getUserProfile() } ,
+
+            ::onGetUserProfileSuccess,
+            ::onGetUserProfileError
+        )
+    }
+
+    private fun onGetUserProfileSuccess (userDetails: UserDetails) {
+        val result= userDetails.toUIState()
+        updateState { it.copy(user = result, fullName = result.fullName, phoneNumber = result.phoneNumber ) }
+    }
+
+    private fun onGetUserProfileError (errorState: ErrorState) {
+        println(errorState)
+    }
+
+    override fun onFullNameChanged(username: String) {
+        updateState { it.copy(fullName = username,isButtonEnabled = true,) }
+    }
+
+    override fun onPhoneNumberChanged(phone: String) {
+        updateState { it.copy(phoneNumber = phone,isButtonEnabled = true,) }
```

Comment:
Where is the validation? I can't see any validation for the phone or name here, neither in the use case. It's better to have validation even if we are aware that backend validation exists, to prevent making too many requests